### PR TITLE
add grace period for evacuate

### PR DESCRIPTION
**Pull request by pweil-**
Fixes https://github.com/openshift/origin/issues/4911

@fabianofranz @deads2k @derekwaynecarr @pmorie @eparis @smarterclayton

### CODE REVIEW

File: pkg/cmd/admin/node/evacuate.go [hunk 1]
```diff
@@ -25,6 +26,8 @@ func (e *EvacuateOptions) AddFlags(cmd *cobra.Command) {
 
 	flags.BoolVar(&e.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
 	flags.BoolVar(&e.Force, "force", false, "Delete pods not backed by replication controller. Optional param for --evacuate")
+	flags.Int64Var(&e.Grace, "grace", 30, "Grace period (seconds) for pods being deleted. Optional param for --evacuate")
```
Comment (smarterclayton) [inline pkg/cmd/admin/node/evacuate.go]: Should be identical to upstream, i.e. `grace-period`

Comment (pweil-) [inline pkg/cmd/admin/node/evacuate.go]: updated name and vars

**Comment by smarterclayton** [comment]
Test case please.

**Comment by pweil-** [comment]
> Test case please.

Added test that expected flags are registered and have expected default values.  Refactored making delete options to its own method so I can test that it is created correctly.  We could probably take this further by refactoring NodeOptions into an interface so commands could mock the behavior and test the expected calls are created.  Perhaps that's why other commands in the package don't have tests at the moment.

PTAL

File: pkg/cmd/admin/node/evacuate.go [hunk 2]
```diff
@@ -12,19 +12,32 @@ import (
...
+	Force       bool
+	GracePeriod int64
 }
 
 func (e *EvacuateOptions) AddFlags(cmd *cobra.Command) {
 	flags := cmd.Flags()
 
-	flags.BoolVar(&e.DryRun, "dry-run", false, "Show pods that will be migrated. Optional param for --evacuate")
-	flags.BoolVar(&e.Force, "force", false, "Delete pods not backed by replication controller. Optional param for --evacuate")
+	flags.BoolVar(&e.DryRun, flagDryRun, defaultDryRun, "Show pods that will be migrated. Optional param for --evacuate")
```
Comment (deads2k) [inline pkg/cmd/admin/node/evacuate.go]: Using the current value of `e.DryRun` makes it possible for different composers to choose different defaults.

**Comment by deads2k** [comment]
Optional nit.  lgtm.

File: pkg/cmd/admin/node/evacuate.go (hunk 2 above)
Comment (pweil-) [inline pkg/cmd/admin/node/evacuate.go]: added New* method in the same pattern as most other commands and changed the manage node command to use it.  Updated tests.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**